src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -140,7 +140,7 @@ public String getScript() {
         FilePath resultFile = c.getResultFile(ws);
         FilePath controlDir = c.controlDir(ws);
         if (capturingOutput) {
-            cmd = String.format("pid=$$; { while ps -o pid -p $pid | grep -q $pid && [ -d '%s' -a \\! -f '%s' ]; do touch '%s'; sleep 3; done } & jsc=%s; %s=$jsc '%s' > '%s' 2> '%s'; echo $? > '%s.tmp'; mv '%s.tmp' '%s'; wait",
+            cmd = String.format("pid=$$; { while ps -o pid | grep -q \"^\\s*$pid$\" && [ -d '%s' -a \\! -f '%s' ]; do touch '%s'; sleep 3; done } & jsc=%s; %s=$jsc '%s' > '%s' 2> '%s'; echo $? > '%s.tmp'; mv '%s.tmp' '%s'; wait",
```

Comment:
To get this to work on csh/tcsh we'd need to use `\"^\\s*$pid\"\'$\'` (becomes `"^\s*$pid"'$'` after removing Java's escaping) because in those shells things like `echo "$"` throw an error.

Do we care about csh/tcsh?

Reply:
I don't *think* so - @jglick might have other thoughts. 

But is *grep* guaranteed to be supported?

Reply:
This is making me twitch a bit.  Which docker images did we test with?

Reply:
No idea about whether it's _guaranteed_ to be supported on all platforms, but it is at least included in BusyBox. I'll look around for a definitive answer.

Reply:
> Which docker images did we test with?

jenkinsci/slave with both alpine (3.19-1-alpine, before the procops fix) and non-alpine (3.20-1) tags.

Reply:
> is grep guaranteed to be supported

Someone could always have a custom arch distro that specifically excludes grep, but I think it is pretty unlikely. The best info I could find about where `grep` is included by default is https://unix.stackexchange.com/questions/37064/which-are-the-standard-commands-available-in-every-linux-based-distribution. Based on that, and the fact that our use is compatible with BusyBox as well, I think it is pretty safe to count on grep being available, unless there are some new popular container OS's that are leaving it out entirely.